diff --git a/panda/src/common.c b/panda/src/common.c
--- a/panda/src/common.c
+++ b/panda/src/common.c
@@ -152,6 +152,8 @@ target_ulong panda_current_asid(CPUState *cpu) {
   #elif defined(TARGET_LOONGARCH)
     CPUArchState *env = cpu_env(cpu);
     return (env->CSR_ASID & 0xff);
+  #elif defined(TARGET_TRICORE)
+    return 0;
   #else
   #error "panda_current_asid() not implemented for target architecture."
     return 0;
diff --git a/panda/src/panda_arch.c b/panda/src/panda_arch.c
--- a/panda/src/panda_arch.c
+++ b/panda/src/panda_arch.c
@@ -31,6 +31,8 @@ bool panda_in_kernel_mode(const CPUState *cpu) {
     // Both supervisor (S) and machine (M) modes are considered kernel mode
     // Check the privilege bits in mstatus CSR
     return (env->priv >= PRV_S);
+#elif defined(TARGET_TRICORE)
+    return false;
 #else
 #error "panda_in_kernel_mode() not implemented for target architecture."
     return false;
@@ -148,6 +150,8 @@ target_ulong panda_current_ksp(CPUState *cpu) {
     // For how stack works in RISC-V, see:
     // https://github.com/riscv/riscv-elf-psabi-doc/blob/master/riscv-elf.md
     return env->gpr[2];
+#elif defined(TARGET_TRICORE)
+    return env->gpr_a[10];
 #else
 #error "panda_current_ksp() not implemented for target architecture."
     return 0;
@@ -184,6 +188,8 @@ target_ulong panda_current_sp(const CPUState *cpu) {
 #elif defined(TARGET_RISCV)
     // For RISC-V, stack pointer is x2/sp
     return env->gpr[2];
+#elif defined(TARGET_TRICORE)
+    return env->gpr_a[10];
 #else
 #error "panda_current_sp() not implemented for target architecture."
     return 0;
@@ -222,6 +228,8 @@ target_ulong panda_get_retval(const CPUState *cpu) {
 #elif defined(TARGET_RISCV)
     // RISC-V uses a0 (x10) for return value
     return env->gpr[10];
+#elif defined(TARGET_TRICORE)
+    return env->gpr_d[2];
 #else
 #error "panda_get_retval() not implemented for target architecture."
     return 0;
@@ -248,6 +256,8 @@ void panda_set_retval(CPUState *cpu, target_ulong value){
 #elif defined(TARGET_RISCV)
     // RISC-V uses a0 (x10) for return value
     env->gpr[10] = value;
+#elif defined(TARGET_TRICORE)
+    env->gpr_d[2] = value;
 #else
 #error "panda_get_retval() not implemented for target architecture."
 #endif
@@ -290,6 +300,9 @@ void panda_set_retval(CPUState *cpu, target_ulong value){
     #define GPR(x) env->gpr[x]
                         // a7, a0    a1    a2    a3    a4    a5    a6
     target_ulong regs[] = {17, 10, 11, 12, 13, 14, 15, 16};
+#elif defined(TARGET_TRICORE)
+    #define GPR(x) env->gpr_d[x]
+    target_ulong regs[] = {2, 4, 5, 6, 7};
 
 #endif
